src/software/ai/navigator/obstacle/robot_navigation_obstacle_factory.cpp

```diff
@@ -112,6 +112,32 @@ RobotNavigationObstacleFactory::createStaticObstaclesFromMotionConstraint(
             // AVOID_BALL_PLACEMENT_INTERFERENCE is not handled by this obstacle factory
             // since it's a dynamic obstacle
             break;
+        case TbotsProto::MotionConstraint::FRIENDLY_GOAL:
+        {
+            const Rectangle &friendly_goal = field.friendlyGoal();
+
+            // Reduce the size of the goal obstacle slightly to avoid the goalie
+            // appear to be inside the obstacle if it is touching one of the goal
+            // walls (e.g. due to noisy vision).
+            const double goal_obstacle_radius = ROBOT_MAX_RADIUS_METERS - 0.01;
+
+            // Top goal post
+            obstacles.push_back(std::make_shared<GeomObstacle<Polygon>>(
+                Polygon::fromSegment(Segment(friendly_goal.posXPosYCorner(),
+                                             friendly_goal.negXPosYCorner()),
+                                     0, goal_obstacle_radius)));
+            // Bottom goal post
+            obstacles.push_back(std::make_shared<GeomObstacle<Polygon>>(
+                Polygon::fromSegment(Segment(friendly_goal.posXNegYCorner(),
+                                             friendly_goal.negXNegYCorner()),
+                                     0, goal_obstacle_radius)));
+            // Left goal wall
+            obstacles.push_back(std::make_shared<GeomObstacle<Polygon>>(
+                Polygon::fromSegment(Segment(friendly_goal.negXPosYCorner(),
+                                             friendly_goal.negXNegYCorner()),
+                                     goal_obstacle_radius)));
```

Comment:
how come you can omit the third argument here?

Reply:
Good question. To avoid having to update every existing usecase of `Polygon::fromSegment` from having 1 radius argument to having 2 arguments, I decided to overload the function with two signatures, one takes 2 arguments and has the full implementation of the function
```cpp
    static Polygon fromSegment(const Segment& segment, double length_radius,
                               double width_radius);
```
and another which takes only 1 radius argument (similar to the one that was already in the code)
```cpp
    static Polygon fromSegment(const Segment& segment, double radius);
```
To avoid duplication of code, the method signature which takes 1 argument basically calls the signature that takes two arguments but with the same value for both `length_radius` and `width_radius`.
E.g. `fromSegment(Segment(), 2.5)` Calls `fromSegment(Segment(), 2.5, 2.5)`. C++ (and most other languages) is smart enough two differentiate the two implementations based on how many arguments you use to call the function.